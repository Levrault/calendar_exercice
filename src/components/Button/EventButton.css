.button {
  background: transparent;
  border: 0;
  cursor: pointer;
  display: flex;
  flex: 1;
  justify-content: center;
  margin: $space-s 0;
  opacity: .5;
  outline: none;
  transition: all .5s;

  &.active,
  &:hover {
    opacity: 1;
  }
}
