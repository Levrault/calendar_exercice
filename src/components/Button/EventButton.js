import React from 'react';
import styles from './EventButton.css';

/**
 * Primary button
 * @param {bool} active
 * @param {node} children
 * @returns {node}
 */
const EventButton = ({ children, ...rest }) => (
  <button {...rest} className={styles.button} >
    {children}
  </button>
);

export default EventButton;
