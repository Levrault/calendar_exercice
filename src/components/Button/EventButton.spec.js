import React from 'react';
import renderer from 'react-test-renderer';
import EventButton from './EventButton';

describe('EventButton component', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <EventButton type="button">
        <i className="material-icons">clear</i>
      </EventButton>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when active', () => {
    const component = renderer.create(
      <EventButton type="button" active>
        <i className="material-icons">clear</i>
      </EventButton>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
