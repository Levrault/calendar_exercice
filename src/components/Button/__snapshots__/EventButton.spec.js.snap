// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventButton component should render correctly 1`] = `
<button
  className=""
  type="button"
>
  <i
    className="material-icons"
  >
    clear
  </i>
</button>
`;

exports[`EventButton component should render correctly when active 1`] = `
<button
  className="undefined"
  type="button"
>
  <i
    className="material-icons"
  >
    clear
  </i>
</button>
`;
